internal/future/statestore/redis.go

```diff
@@ -119,7 +120,70 @@ func (rb *redisBackend) DeindexTicket(context.Context, string, []string) error {
 	return status.Error(codes.Unimplemented, "not implemented")
 }
 
-// FilterTickets returns the Ticket ids for the Tickets meeting the specified filtering criteria.
-func (rb *redisBackend) FilterTickets(context.Context, []pb.Filter) ([]string, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+// FilterTickets returns the Ticket ids and required attribute key-value pairs for the Tickets meeting the specified filtering criteria.
+// map[ticket.Id]map[attributeName][attributeValue]
+func (rb *redisBackend) FilterTickets(filters []*pb.Filter) (map[string]map[string]int64, error) {
+	redisConn := rb.redisPool.Get()
+
+	defer func() {
+		if err := redisConn.Close(); err != nil {
+			redisLogger.WithError(err).Error("Failed to close Redis connection")
+		}
+	}()
+
+	// A map[attribute]map[playerID]value
+	attributeToTickets := make(map[string]map[string]int64)
+	// A set of playerIds that satisfies all filters
+	idSet := make([]string, 0)
+
+	// For each filter, do a range query to Redis on Filter.Attribute
+	for i, filter := range filters {
+		startTime := time.Now()
+		// Time Complexity O(logN + M), where N is the number of elements in the attribute set
+		// and M is the number of entries being returned.
+		idToValue, err := redis.Int64Map(redisConn.Do("ZRANGEBYSCORE", filter.Attribute, filter.Min, filter.Max, "WITHSCORES"))
```

Comment:
So if the responses exceed max here - we silently only process up to max right? how is that surfaced to the user? Should we log that we only processed so many? is there a way to get the total count so we can tell how many were missed? I ll create an issue to track the work to figure how this needs to be surfaces to the caller.

Reply:
Redis does not come with a default maximum number of entries being returned for each command. As long as we are executing the command a [`LIMIT`](https://redis.io/commands/zrangebyscore) option, Redis will return all results to the client.

Reply:
So are we using the LIMIT then? theres ZCOUNT too - the question is if do we really want to support this unbounded.

Reply:
Added a TODO to fix later.